core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -480,6 +480,34 @@ public static boolean validFileNameExcludingAstrix(String fileName) {
     }
 
     /**
+     * This does a very basic pass at validating that a header's value contains only expected characters according to RFC-5987, and those
+     * that it references.
+     * <p>
+     * https://tools.ietf.org/html/rfc5987
+     * <p>
+     * This is only expected to be used for assertions. The idea is that only readable US-ASCII characters are expected; the rest must be
+     * encoded with percent encoding, which makes checking for a valid character range very simple.
+     *
+     * @return {@code true} if the {@code value} is not obviously wrong.
+     */
+    public static boolean validHeaderValue(String value) {
+        if (hasLength(value) == false) {
+            return false;
+        }
+
+        for (int i = 0; i < value.length(); i++) {
+            char c = value.charAt(i);
+
+            // 32 = ' ' (31 = unit separator); 126 = '~' (127 = DEL)
+            // http://www.asciitable.com/ in case you need a reference
```

Comment:
This comment can be removed, it's not authoritative, it might get broken, and it's easily discovered via search. What really matters is the RFC link which you've provided above already. :smile: